main/src/library/Result.flix

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2016 Liam Palmer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Result {
+
+    ///
+    /// Returns `true` iff `r` is `Ok(v)`.
+    ///
+    def isOk[a, b](r: Result[a, b]): Bool = match r with {
```

Comment:
Super general comment: For `Result` the type parameters should be named `t` and `e` (to remind us that `t` is the "useful" type and `e` is the error type). So if you could sub. a and b for t and e.